perl/lib/JSON/Structure/InstanceValidator.pm

```diff
@@ -0,0 +1,1756 @@
+    }
+    
+    $self->{current_depth}--;
+}
+
+sub _validate_type {
+    my ($self, $value, $type, $schema, $path, $schema_path) = @_;
+    
+    # Validate based on type
+    if ($type eq 'null') {
+        $self->_validate_null($value, $path, $schema_path);
+    }
+    elsif ($type eq 'boolean') {
+        $self->_validate_boolean($value, $path, $schema_path);
+    }
+    elsif ($type eq 'string') {
+        $self->_validate_string($value, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'number' || $type eq 'float' || $type eq 'double' || $type eq 'float8' || $type eq 'decimal') {
+        $self->_validate_number($value, $type, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'integer' || exists $INT_RANGES{$type}) {
+        $self->_validate_integer($value, $type, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'int128' || $type eq 'uint128') {
+        $self->_validate_big_integer($value, $type, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'object') {
+        $self->_validate_object($value, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'array') {
+        $self->_validate_array($value, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'set') {
+        $self->_validate_set($value, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'map') {
+        $self->_validate_map($value, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'tuple') {
+        $self->_validate_tuple($value, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'choice') {
+        $self->_validate_choice($value, $schema, $path, $schema_path);
+    }
+    elsif ($type eq 'any') {
+        # Any type accepts all values
+    }
+    elsif ($type eq 'date') {
+        $self->_validate_date($value, $path, $schema_path);
+    }
+    elsif ($type eq 'time') {
+        $self->_validate_time($value, $path, $schema_path);
+    }
+    elsif ($type eq 'datetime') {
+        $self->_validate_datetime($value, $path, $schema_path);
+    }
+    elsif ($type eq 'duration') {
+        $self->_validate_duration($value, $path, $schema_path);
+    }
+    elsif ($type eq 'uuid') {
+        $self->_validate_uuid($value, $path, $schema_path);
+    }
+    elsif ($type eq 'uri') {
+        $self->_validate_uri($value, $path, $schema_path);
+    }
+    elsif ($type eq 'binary') {
+        $self->_validate_binary($value, $path, $schema_path);
+    }
+    elsif ($type eq 'jsonpointer') {
+        $self->_validate_jsonpointer($value, $path, $schema_path);
+    }
+    else {
+        $self->_add_error(
+            INSTANCE_TYPE_UNKNOWN,
+            "Unknown type: $type",
+            $path,
+            $schema_path
+        );
+    }
+}
+
+sub _check_type {
+    my ($self, $value, $type) = @_;
+    
+    if ($type eq 'null') {
+        return !defined $value || (ref($value) eq '' && $value eq 'null');
+    }
+    elsif ($type eq 'boolean') {
+        return JSON::PP::is_bool($value) || 
```

Comment:
use Scalar::Util 'blessed';
blessed $x && $x ->isa($class); for applicable json boolean types